src/Microsoft.AspNetCore.Hosting/Internal/WebHostOptions.cs

```diff
@@ -22,6 +22,7 @@ public WebHostOptions(IConfiguration configuration)
             }
 
             Application = configuration[WebHostDefaults.ApplicationKey];
+            EnvironmentApplication = configuration[WebHostDefaults.EnvironmentApplicationKey];
```

Comment:
How would MVC want to use it? I think this feature is used because MVC needed it?